config/config.go

```diff
@@ -62,26 +81,147 @@ func TryLoadRootConfig(fromdir string) (cfg hcl.Config, configpath string, found
 		}
 		fromdir = parent
 	}
-	return hcl.Config{}, "", false, nil
+	return nil, "", false, nil
 }
 
-// IsStack returns true if the given directory is a stack, false otherwise.
-func IsStack(rootdir, dir string) (bool, error) {
+func LoadTree(rootdir string, cfgdir string) (*Tree, error) {
+	return loadTree(rootdir, cfgdir, nil)
+}
+
+func (tree *Tree) Rootdir() string {
+	return tree.rootdir
+}
+
+func (tree *Tree) IsStack() bool {
+	return tree.Root.Stack != nil
+}
+
+func (tree *Tree) Stacks() []*Tree {
+	var stacks []*Tree
+	if tree.IsStack() {
+		stacks = append(stacks, tree)
+	}
+	for _, children := range tree.Children {
+		stacks = append(stacks, children.Stacks()...)
+	}
+
+	return stacks
+}
+
+func (tree *Tree) Lookup(path project.Path) (*Tree, bool) {
+	pathstr := path.String()
+	if len(pathstr) == 0 || pathstr[0] != '/' {
```

Comment:
hmm doesn't project.Path already guarantees these invariants ? maybe it should have a Valid method to check if the path is sane (because in Go it is hard to guarantee invariants on types etc). Feel free to disregard and move on, not important.

Reply:
There's this:
https://github.com/mineiros-io/terramate/blob/main/project/project.go#L43

but making project.Path a struct instead of a string alias would enforce even more.